libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -470,54 +366,61 @@ - (NSString*)uniqueUserAccountIdentifier {
     return identifier;
 }
 
+-(SFOAuthCredentials *)oauthCredentials {
```

Comment:
...I mean, I kind of get it.  `SFUserAccountManager` ends up owning a lot of the bootstrapping properties for a credentials object, including our—valid? seems reasonable—desire to link the credentials identity with its eventual associated user account.  In all other ways, though, it feels like a process that should rightfully be owned by `SFAuthenticationManager`.

I also feel like those bootstrapping properties—`oauthCompletionUrl`, `loginHost`, `oauthClientId`—would be more rightfully owned by `SalesforceSDKManager`.  Indeed, we have convenience public API on the SDK manager that already sets a lot of these, passing them through to the user account manager.

In either case, I'd prefer that `SFAuthenticationManager` would be the one creating the credentials, reaching out to one of those two entities for the proper bootstrapping values.